src/logion/services/transaction.extractor.ts

```diff
@@ -46,40 +51,60 @@ export class TransactionExtractor {
             .build()
     }
 
-    private async extractTransaction(extrinsic: JsonExtrinsic, type: ExtrinsicType, index: number): Promise<Transaction> {
+    private async extractTransactions(extrinsic: JsonExtrinsic, type: ExtrinsicType, index: number): Promise<Transaction[]> {
+        const transactions: Transaction[] = [];
         let from: string = this.from(extrinsic);
         let transferValue: bigint = 0n;
         let to: string | undefined = undefined;
 
-        switch (type) {
-            case ExtrinsicType.TRANSFER:
-                transferValue = this.transferValue(extrinsic)
-                to = this.to(extrinsic)
-                break
-            case ExtrinsicType.TRANSFER_FROM_RECOVERED:
-                const account = this.extrinsicDataExtractor.getAccount(extrinsic);
-                if (account) {
-                    from = account
-                }
-                const call = this.extrinsicDataExtractor.getCall(extrinsic)
-                transferValue = this.transferValue(call)
-                to = this.to(call)
-                break
+        if(type ===  ExtrinsicType.TRANSFER) {
```

Comment:
It's a matter of personal preference, but IMHO switch in way more readable that this chain of `if-else if`.